src/eval/merge.rs

```diff
@@ -367,6 +367,15 @@ pub fn merge(
         // Merge put together the fields of records, and recursively merge
         // fields that are present in both terms
         (Term::Record(mut r1), Term::Record(mut r2)) => {
+            // While it wouldn't be impossible to merge records with sealed tails,
+            // working out how to do so in a "sane" way that preserves parametricity
+            // is non-trivial. It's also not entirely clear that this is something
+            // users will generally have reason to do, so in the meantime we've
+            // decided to just prevent this entirely
+            if let Some(record::SealedTail { label, .. }) = r1.sealed_tail.or(r2.sealed_tail) {
+                return Err(EvalError::BlameError(label, std::mem::take(call_stack)));
```

Comment:
[This comment](https://github.com/tweag/nickel/pull/949/files#r1033797621) also applies here re: explicitly setting the label's `tag`.